packages/client/src/views/AdvancedSearch/RemoveBookmarkModal.tsx

```diff
@@ -0,0 +1,144 @@
+  setNotificationStatus: (status: NOTIFICATION_STATUS) => void
+  setNotificationMessages: (notificationMessage: string) => void
+}
+
+interface IRemoveBookmarkData {
+  removeBookmarkedAdvancedSearch: BookMarkedSearches
+}
+
+export function RemoveBookmarkAdvancedSearchModal({
+  showRemoveBookmarkModal,
+  toggleRemoveBookmarkModal,
+  setNotificationStatus,
+  setNotificationMessages
+}: IRemoveBookmarkModalProps) {
+  const intl = useIntl()
+  const dispatch = useDispatch()
+  const { searchId, ...rest } = useSelector(getPartialState)
+  const userDetails = useSelector(getUserDetails)
+
+  const [removeAdvancedSearchResultBookmark] = useMutation<
+    IRemoveBookmarkData,
+    MutationRemoveBookmarkedAdvancedSearchArgs
+  >(REMOVE_ADVANCED_SEARCH_RESULT_BOOKMARK_MUTATION, {
+    onError() {
+      setNotificationMessages(
+        intl.formatMessage(
+          messagesSearch.advancedSearchBookmarkErrorNotification
+        )
+      )
+      setNotificationStatus(NOTIFICATION_STATUS.ERROR)
+    },
+    onCompleted() {
+      setNotificationMessages(
+        intl.formatMessage(
+          messagesSearch.removedAdvancedSearchBookmarkSuccessNotification
+        )
+      )
+      setNotificationStatus(NOTIFICATION_STATUS.SUCCESS)
+    }
+  })
+
+  const removeAdvancedSearchBookmarkHandler = async () => {
+    const mutatedData = await removeAdvancedSearchResultBookmark({
+      variables: {
+        removeBookmarkedSearchInput: {
+          userId: String(userDetails?.userMgntUserID),
+          searchId: String(searchId)
+        }
+      }
+    })
+    if (mutatedData.data) {
+      dispatch(
+        modifyUserDetails({
+          ...userDetails,
+          searches: mutatedData.data.removeBookmarkedAdvancedSearch
+            .searchList as GQLBookmarkedSeachItem[]
+        })
+      )
+      dispatch(setAdvancedSearchParam({ searchId: EMPTY_STRING, ...rest }))
+    }
+  }
+  return (
+    <>
+      <ResponsiveModal
+        id="removeBookmarkModal"
+        show={showRemoveBookmarkModal}
+        title={intl.formatMessage(
+          messagesSearch.removeBookmarkAdvancedSearchModalTitle
+        )}
+        autoHeight={true}
+        actions={[
+          <Button
+            type="tertiary"
+            id="cancel"
+            key="cancel"
+            onClick={toggleRemoveBookmarkModal}
+          >
+            {intl.formatMessage(buttonMessages.cancel)}
+          </Button>,
+          <Button
+            type="primary"
+            id="remove_advanced_search_bookmark"
+            onClick={async () => {
+              setNotificationMessages(
+                intl.formatMessage(
+                  messagesSearch.removeAdvancedSearchBookmarkLoadingNotification
+                )
+              )
+              setNotificationStatus(NOTIFICATION_STATUS.IN_PROGRESS)
+              toggleRemoveBookmarkModal()
+              await removeAdvancedSearchBookmarkHandler()
+            }}
+            disabled={false}
```

Comment:
We should probably use the online status here instead

Reply:
Yeah